### PR TITLE
fix #2012, cmake find vulkan issue

### LINKED ISSUE #2012: vulkun无法编译
-- CMAKE_INSTALL_PREFIX = /home/xxx/Documents/ncnn/build/install
-- The C compiler identification is GNU 5.4.0
-- The CXX compiler identification is GNU 5.4.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Target arch: x86
CMake Warning at CMakeLists.txt:116 (message):
  The submodules were not downloaded! NCNN_VULKAN_ONLINE_SPIRV will be turned
  off.

CMake Warning at CMakeLists.txt:117 (message):
  Please update submodules with "git submodule update --init" and try again.

-- Found glslangValidator: GLSLANGVALIDATOR_EXECUTABLE-NOTFOUND
-- Try OpenMP C flag = [-fopenmp]
-- Performing Test OpenMP_FLAG_DETECTED
-- Performing Test OpenMP_FLAG_DETECTED - Success
-- Try OpenMP CXX flag = [-fopenmp]
-- Performing Test OpenMP_FLAG_DETECTED
-- Performing Test OpenMP_FLAG_DETECTED - Success
-- Found OpenMP: -fopenmp  
CMake Error at src/CMakeLists.txt:175 (find_package):
  By not providing "FindVulkan.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "Vulkan", but
  CMake did not find one.

  Could not find a package configuration file provided by "Vulkan" with any
  of the following names:

    VulkanConfig.cmake
    vulkan-config.cmake

  Add the installation prefix of "Vulkan" to CMAKE_PREFIX_PATH or set
  "Vulkan_DIR" to a directory containing one of the above files.  If "Vulkan"
  provides a separate development package or SDK, be sure it has been
  installed.

-- Configuring incomplete, errors occurred!
See also "/home/xxx/Documents/ncnn/build/CMakeFiles/CMakeOutput.log".

我在编译的时候除了这个问题，想问问可以怎么解决？是cmake版本问题还是缺了什么的原因？我是按照教程直接wget的vulkun，cmake版本是3.5.1

### PR DESCRIPTION
**Pull request by zchrissirhcz**
增加了一段友好的输出，提示用户设定 `VULKAN_SDK` 环境变量

并且给CMAKE_MODULE_PATH追加路径，这对于 cmake < 3.7.0 的版本（譬如广大的ubuntu 16.04用户的apt装的cmake 3.5.1），是需要的。从 3.7.0开始cmake自带了FindVulkan.cmake。

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -172,7 +172,20 @@ if(NCNN_OPENMP)
...
+    if(NOT Vulkan_FOUND)
+        if(DEFINED ENV{VULKAN_SDK})
+            list(APPEND CMAKE_MODULE_PATH "$ENV{VULKAN_SDK}/../source/VulkanTools/cmake")
+        else()
+            message(FATAL_ERROR "!! CMake didn't find Vulkan. One of the following helps:\n"
+                "1. export env var VULKAN_SDK before cmake, e.g.\n"
+                "    export VULKAN_SDK=~/soft/vulkansdk/1.2.148.0/x86_64\n"
+                "2. use pkg-config in CMakeLists.txt"
+            )
+        endif()
```
Comment (nihui) [inline src/CMakeLists.txt:185]: 这段是 Linux specific 的？
windows 和 macos 是如何的？

Comment (zchrissirhcz) [reply src/CMakeLists.txt:185]: 现在手头没有设备，晚上测试一下。

**Comment by tencent-adm** [comment]
[CLA assistant check](https://cla.opensource.tencent.com/Tencent/ncnn?pullRequest=2017) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla.opensource.tencent.com/Tencent/ncnn?pullRequest=2017) before we can accept your contribution.<br/>**2** out of **3** committers have signed the CLA.<br/><br/>:white_check_mark: nihui<br/>:white_check_mark: zchrissirhcz<br/>:x: restyled-commits<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.tencent.com/check/Tencent/ncnn?pullRequest=2017) it.</sub>

**Comment by zchrissirhcz** [comment]
macos的vulkansdk里，没找到FindVulkan.cmake。因此，对于macos上使用<3.7.0的cmake情况，输出信息是让用户升级cmake。

windows和linux则分别设定了不同的CMAKE_MODULE_PATH append项。

**Comment by nihui** [comment]
感谢

**Pull request merged**